### PR TITLE
sql-query: performance improvements by reducing http calls

### PR DESCRIPTION
**Pull request by chassing**
* use bulk request to fetch "/aws/rds-defaults-1.yml" resources all at once
* use state.ls instead of state.exists for query states

## old
QUERY_COUNT * "get rds default resource" + QUERY_COUNT * (state.exists + state.get)

674 + 674 * 2 = 2022 http calls

## new
"get all rds default resources" + QUERY_COUNT * state.get + state.ls

1 + 674 + 1 = 676 http calls

APPSRE-6546

### CODE REVIEW

File: reconcile/utils/gql.py [hunk 1]
```diff
@@ -188,6 +188,22 @@ def get_resource(self, path: str) -> dict[str, Any]:
...
+                path
+                content
+                sha256sum
+            }
+        }
+        """
+
+        # Do not validate schema in resources since schema support in the
+        # resources is not complete.
+        resources = self.query(query, {"schema": schema}, skip_validation=True)
```
Comment (fishi0x01) [inline reconcile/utils/gql.py:204]: Not sure I understand this. How can we query sth with gql that is not defined in the schema?

Comment (chassing) [reply reconcile/utils/gql.py:204]: This is a very good question! Actually, I've copied and pasted the original [get_resource](https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/gql.py#L166) method and adapted it 🤪

Comment (fishi0x01) [reply reconcile/utils/gql.py:204]: @rporres do you know hot this is possible?

Comment (chassing) [reply reconcile/utils/gql.py:204]: Or maybe @geoberle can explain that?

Comment (rporres) [reply reconcile/utils/gql.py:204]: Hmmm, there's no GraphQL schema for anything under `app-interface/resources`, only JSON schema. GraphQL schema only refers to objects under `data`. Resources are usually free form YAML objects and we only add validation in a few of them.

File: reconcile/utils/gql.py [hunk 2]
```diff
@@ -188,6 +188,22 @@ def get_resource(self, path: str) -> dict[str, Any]:
 
         return resources[0]
 
+    def get_resources_by_schema(self, schema: str) -> list[dict[str, str]]:
```
Comment (rporres) [inline reconcile/utils/gql.py:191]: we love docstrings and we are sad when new code doesn't add them 😢

**Review by rporres — APPROVED** [review]
This is cool! Please add documentation for the new public methods you're adding 🙏

**Pull request merged**